docs/EXTENDED-ARCHIVES.md

```diff
@@ -0,0 +1,428 @@
+Etcher Extended Archives
+========================
+
+Etcher extended archives are archive files including an image and a special
+metadata directory including information that describes the image, device type,
+and vendor in detail, enabling Etcher to make use of such information to
+deliver advanced features and a more pleasant experience to the user.
+
+Entities
+--------
+
+### Vendor
+
+A vendor represents a manufacturer of a device type and potential publisher.
+
+*Some examples of image vendors are: Beagleboard, Raspberry Pi Foundation,
+Intel, and ODroid.*
+
+### Publisher
+
+A publisher represents an entity that maintains and publishes official images.
+
+*Some examples of image publishers are: Tizen, RetroPie, Resin.io, and CoreOS.*
+
+### Device
+
+A device represents a specific device type manufactured by a vendor.
+
+*Some examples of devices are: Beaglebone Black, Beaglebone Green, Intel
+Edison, and Raspberry Pi 3.*
+
+### Client
+
+A client represents the program making use of the information about the other
+entities.
+
+*At the moment of this writing, the only existent client is the Etcher application.*
+
+File format
+-----------
+
+The file format should have the following characteristics:
+
+- Streamability: The extended archive allows the client to consume all its
```

Comment:
IMHO the (possibly multi-gigabyte) disk-image itself should be streamable, but I don't think the metadata itself needs to be streamable (since it'll be small enough to buffer in RAM).
I still wonder if we ought to do something like the RPF does with it's NOOBS images regarding the separation of metadata (including image logos etc.) and (compressed) disk-images. And then for 'downloadable' (offline) images we could have a zip including both the metadata and the image-file.
Given that `.zip` is the lowest-common-denominator compression format, but it's not streamable, I think it _might_ make sense to have separate formats for the offline-download images (where we only want a single file, compressed in .zip), and the online-streamable images (where we want streamability, but there's no problems in having multiple separate files, as long as they're all reachable from a "starting URL"). Does that make sense?

Hmmm, maybe something like:
On the server: `metadata.json`, `metadata_XX.zip`, `diskimage_XX.img.xz` (with `metadata.json` being a file at a well-known/stable URL, which would be what Etcher 'consumes', which then contains links to `metadata_XX.zip` and `diskimage_XX.img.xz`)
In the downloaded `diskimage_XX.zip`: `diskimage_XX.img`, `.meta/metadata.json`, `.meta/<other metadata files pointed to by metadata.json>`

This makes it easy for publishers to start enhancing their current `diskimage_XX.zip` files to support the `.meta/` metadata directory (i.e. only a small change to their current workflow), and then in the future if they want to they can also start supporting online-streamable Etcher images. And it should be easy for us to supply them with a script which converts an offline-zipfile into the separate online-streamable files (and also the inverse, if desired).